policy/compressor.go

```diff
@@ -0,0 +1,127 @@
+package policy
+
+import (
+	"sync"
+
+	"go.uber.org/zap"
```

Comment:
On second thought, we should probably have this discussion later. I'll switch to `xlog` here.